flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/reader/deserializer/KafkaValueOnlyDeserializerWrapper.java

```diff
@@ -62,8 +75,15 @@ public void open(DeserializationSchema.InitializationContext context) throws Exc
                                     deserializerClass.getName(),
                                     Deserializer.class,
                                     getClass().getClassLoader());
+
+            if (config.isEmpty()) {
+                return;
+            }
+
             if (deserializer instanceof Configurable) {
                 ((Configurable) deserializer).configure(config);
+            } else {
```

Comment:
Can you also add a test to verify the configuration of a de/serializer not implementing `Configurable`?

Reply:
Do you mean if it did implement Configurable, then `Deserializer.configure` will be called?

Reply:
Currently, we have the following scenarios:

1. `De/Serializer implements Configurable` I would **only** call `Configurable.configure`
2. Does not implement Configurable but a configuration is given call `De/Serializer.configure`
3. No configuration is given we call `De/Serializer.configure` with an empty map?

Probably you have to write tests for all of them although I am not fully sure about the last. I guess calling `configure` with an empty map does not hurt.

Reply:
The first one was there before this PR, it should already be tested. For the third one, either we do nothing like I proposed which might need a test, or we don't change it, which will be one of the first or the second case.